### ISSUE TITLE
Upsert Compaction: Schema / Index updates

### ISSUE THREAD

**Issue by tibrewalpratik17** [apache/pinot#13494]
I have not yet explored how schema or index updates affect the CRC mismatch issue in Upsert compaction. I'm adding this as a task in the parent issue so we can track it for some time. I will test updating the schema and indexes, and if everything works out fine, we can close this issue.

**Comment by tibrewalpratik17** [comment]
@tarun11Mavani from Uber will help me validate this.

**Comment by tarun11Mavani** [comment]
Sure @tibrewalpratik17 . Assign this to me.

**Comment by tarun11Mavani** [comment]
As part of this analysis, I discovered that when we update the schema or the index on an upsert-realtime table and then use the reload segments API to reload all segments, it doesn't update the CRC of the existing segments. The CRC is stored in `creation.meta` and is not updated when server reload happens. I verified this by monitoring the `creation.meta` file and also compared the segment metadata from server and zookeeper before and after the segment reload.
This behavior ensures that schema and index changes do not adversely affect the Upsert Compaction task, maintaining seamless compaction of older segments.
